pkgs/tools/admin/acme.sh/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub }:
+stdenv.mkDerivation rec {
+  name = "acme.sh-${version}";
+  version = "2.7.8";
+
+  src = fetchFromGitHub {
+    owner = "Neilpang";
+    repo = "acme.sh";
+    rev = version;
+    sha256 = "0zm64z7av63xi7yjhljab2i8q1vx4q1mpcmcm58jm6k4babalxrf";
+  };
+
+  phases = [ "installPhase" ];
+
+  installPhase = ''
+    mkdir -p $out $out/bin $out/lib
+    cp -R $src/* $_
+    ln -s $out/lib/acme.sh $out/bin
```

Comment:
I'm not sure if that belongs to `lib`: `share` subdirectory seems to be more appropriate. Generally, when upstream is not FHS-compliant, it is usually cleaner to just expose a symlink to executable:

```
  installPhase = "mkdir -p $out/bin && ln -s $src/acme.sh $_";
```

Reply:
Did you really meant to link from `$src` here?

Reply:
`$src` is just a fixed-output derivation, so it shouldn't be garbage collected if there are references to it. In this case, repo mostly contains documentation and the shell script itself. Is there a problem with this approach?

Reply:
I'm mostly worried about clutter and increased risk of collisions when it comes to `environment.pathsToLink`.

Reply:
Only it requires to pull two derivations instead of one, but maybe that's ok.

Reply:
There is also `libexec` that is not symlinked by default which also seems appropriate.